source/opt/inline_pass.cpp

```diff
@@ -402,23 +468,117 @@ bool InlinePass::Inline(ir::Function* func) {
   return modified;
 }
 
-bool InlinePass::IsInlinableFunction(const ir::Function* func) {
-  // We can only inline a function if it has blocks.
-  if (func->cbegin() == func->cend())
+bool InlinePass::hasMultipleReturns(ir::Function* func) {
+  bool seenReturn = false;
+  bool multipleReturns = false;
+  for (auto& blk : *func) {
+    auto lii = blk.cend();
+    lii--;
+    if (lii->opcode() == SpvOpReturn || lii->opcode() == SpvOpReturnValue) {
+      if (seenReturn) {
+        multipleReturns = true;
+        break;
+      }
+      seenReturn = true;
+    }
+  }
+  return multipleReturns;
+}
+
+void InlinePass::ComputeStructuredSuccessors(ir::Function* func) {
+  // If header, make merge block first successor.
+  for (auto& blk : *func) {
+    auto mii = blk.end();
+    mii--;
```

Comment:
A little thing:   For iterators, always prefer pre-decrement (or pre-increment) to post-increment (or post-increment).    That way the update is done in-place and the compiler won't have to make a copy of the iterator.